ModelController.js

```diff
@@ -31,6 +31,16 @@ exports.getData = function () {
     return modelConfig.data;
 }
 
+exports.setModelConfig = function (newModelConfig){
+    if(!newModelConfig.views){
+        newModelConfig.views = []
+    }
+    if(!newModelConfig.containers){
+        newModelConfig.containers = []
+    }
+    modelConfig = newModelConfig
+}
```

Comment:
I presume this replaces `exports.setConfiguration()`? If so, might as well remove `setConfiguration()`

Reply:
They are really similar but they aren't the same. Notably `setConfiguration` just updates the fields while intentionally not setting `data` and `setModelConfig` is replacing the whole object and ensures `views` and `containers` are initialized. While I am sure they can eventually be consolidated, its slightly non-trivial to ensure there aren't adverse effects with what the web app is expecting to have happen.

Reply:
I have made an issue so this can be addressed in the future.

Reply:
Gotcha. I didn't see any references to `setConfiguration` in this PR anywhere, so I assumed it wasn't being used anymore. But of course the PR doesn't show the full context of all files, so it's probably still being used.